pkg/query/endpointset.go

```diff
@@ -389,7 +395,20 @@ func (e *EndpointSet) Update(ctx context.Context) {
 				newRef.Close()
 				return
 			}
+			if len(newRef.labelSets()) > 0 {
```

Comment:
Looks its possible to have empty labelSets.

Reply:
Won't this preserve endpoints with empty labels sets even if the selector has a relabel with `keep`?